File: src/main/java/me/chell/samsara/api/widget/WidgetManager.java
package me.chell.samsara.api.widget;

import me.chell.samsara.api.Loadable;
import me.chell.samsara.impl.widget.*;

import java.util.ArrayList;
import java.util.List;

public class WidgetManager implements Loadable {
    private List<Widget> widgets;

    @Override
    public void load() {
        widgets = new ArrayList<>();
        widgets.add(new Watermark());

        for(Widget w : widgets) {
            w.load();
        }
    }

    @Override
    public void unload() {
        for(Widget w : widgets) {
            w.unload();
        }

        widgets = new ArrayList<>();
    }

    public Widget getWidget(String name) {
        for(Widget w : widgets) {
            if(w.getName().equalsIgnoreCase(name)) return w;
        }
        return null;
    }

    public List<Widget> getWidgets() {
        return widgets;
    }
}

File: src/main/kotlin/me/chell/samsara/api/widget/WidgetManager.kt
package me.chell.samsara.api.widget

import me.chell.samsara.api.Loadable
import me.chell.samsara.impl.widget.Watermark

object WidgetManager: Loadable {

    val widgets = mutableListOf<Widget>()

    override fun load() {
        widgets.add(Watermark())

        for(w in widgets) w.load()
    }

    override fun unload() {
        for(w in widgets) w.unload()
        widgets.clear()
    }
}